src/requests/misc.jl

```diff
@@ -5,47 +5,49 @@ end
 function setTraceNotification_notification(params, server::LanguageServerInstance, conn)
 end
 
-function julia_getCurrentBlockRange_request(params::TextDocumentPositionParams, server::LanguageServerInstance, conn)
-    tdpp = params
+function julia_getCurrentBlockRange_request(tdpp::TextDocumentPositionParams, server::LanguageServerInstance, conn)
     doc = getdocument(server, URI2(tdpp.textDocument.uri))
     offset = get_offset(doc, tdpp.position)
     x = getcst(doc)
     loc = 0
-    p1, p2, p3 = 0, x.span, x.fullspan
+
     if typof(x) === CSTParser.FileH
-        (offset > x.fullspan || x.args === nothing) && return Position(get_position_at(doc, p1)...), Position(get_position_at(doc, p2)...), Position(get_position_at(doc, p3)...)
         for a in x.args
-            if loc <= offset < loc + a.fullspan
-                if typof(a) === CSTParser.ModuleH
-                    if loc + a.args[1].fullspan + a.args[2].fullspan < offset < loc + a.args[1].fullspan + a.args[2].fullspan + a.args[3].fullspan
-                        loc0 = loc +  a.args[1].fullspan + a.args[2].fullspan
+            if loc <= offset <= loc + a.span
+                if CSTParser.defines_module(a) # Within module at the top-level, lets see if we can select on of the block arguments
+                    if loc <= offset <= loc + a.args[1].span # Within `module` keyword, so return entire expression
```

Comment:
This would be really helpful, while we still want https://github.com/julia-vscode/julia-vscode/issues/1330 to be solved for great module experience.